charts/k8ssandra-cluster/templates/stargate/stargate.yaml

```diff
@@ -0,0 +1,75 @@
+{{- if .Values.stargate.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-k8ssandra-stargate
+  labels:
+    app: {{ .Release.Name }}-k8ssandra-stargate
+{{ include "k8ssandra-cluster.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.stargate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}-k8ssandra-stargate
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}-k8ssandra-stargate
+    spec:
+      containers:
+        - name: {{ .Release.Name }}-k8ssandra-stargate
+  {{- if eq .Values.stargate.containerImage "auto" }}
+    {{- if eq .Values.stargate.clusterVersion "3.11" }}
+          image: "stargateio/stargate-3_11:v0.0.9"
+    {{- else if eq .Values.stargate.clusterVersion "4.0" }}
+          image: "stargateio/stargate-4_0:v0.0.9"
+    {{- else }}
+          image: "containerImage-auto-did-not-recognize-clusterVersion-{{ .Values.stargate.clusterVersion }}"
+    {{- end }}
+  {{- else }}
+          image: {{ .Values.stargate.containerImage | quote }}
+  {{- end }}
+          ports:
+            - name: graphql
+              containerPort: 8080
+            - name: authorization
+              containerPort: 8081
+            - name: rest
+              containerPort: 8082
+            - name: admin
+              containerPort: 8084
+            - name: metrics
+              containerPort: 8085
+          resources:
+            limits:
+              memory: {{ .Values.stargate.resourceLimits.memory | quote }}
+          env:
+            - name: JAVA_OPTS
+              value: {{ .Values.stargate.javaOpts | quote }}
+            - name: CLUSTER_NAME
+              value: {{ .Values.clusterName | quote }}
+            - name: CLUSTER_VERSION
+              value: {{ .Values.stargate.clusterVersion | quote }}
+            - name: SEED
+              value: "{{ .Values.clusterName }}-seed-service"
+            - name: DATACENTER_NAME
+              value: {{ .Values.datacenterName | quote }}
+            - name: RACK_NAME
+              value: {{ .Values.rackName | quote }}
+            - name: ENABLE_AUTH
+              value: "false"
+          startupProbe:
```

Comment:
I think we may want adjust the probes a bit. Here is some output from a 3-node cluster that I just created:

```
$ kubectl get pods
NAME                                                        READY   STATUS             RESTARTS   AGE
cass-operator-86d4dc45cd-95ght                              1/1     Running            0          2m12s
grafana-deployment-69688df955-8bk6f                         1/1     Running            0          31s
k8ssandra-kube-prometheus-operator-755f6c444-vw2fx          1/1     Running            0          2m12s
prometheus-stargate-test-prometheus-k8ssandra-0             2/2     Running            1          84s
stargate-test-dc1-default-sts-0                             2/2     Running            0          84s
stargate-test-dc1-default-sts-1                             1/2     Running            0          84s
stargate-test-dc1-default-sts-2                             1/2     Running            0          84s
stargate-test-grafana-operator-k8ssandra-57cd775b6f-h686r   1/1     Running            0          85s
stargate-test-k8ssandra-stargate-5c84fc5d5f-gd66s           0/1     CrashLoopBackOff   3          85s
stargate-test-reaper-operator-k8ssandra-cdb4c54b9-n8l56     1/1     Running            0          85s
```
In my experience thus far it seems like it can take upwards of 1 minute for a Cassandra pod to become ready (assuming backup/restore disabled). cass-operator starts Cassandra instances serially. For my 3 node cluster, I would expect it to take around 3 minutes before the C* cluster (i.e., CassandraDatacenter) is ready. And this is with reasonable hardware.

With these settings the, we will only see more restarts of Stargate as the C* cluster size increases. I suggest adding an `initialDelaySeconds` for `startupProbe` with maybe a default value of 30 seconds. See how that does. It might make sense to increase the value based on the size of the C* cluster up to some limit.

It might be worthwhile to experiment with increasing the `periodSeconds` property of `startupProbe`. It defaults to 10 seconds. A higher value might help as well.

Reply:
I discovered while testing on a GKE cluster earlier today that the startup problem may not be available on all clusters. Given this I think we need to only rely on the liveness and readiness probes for now.

We need a couple other changes for the `livenessProbe`:

* Set a default value for `initialDelaySeconds`
* Make `initialDelaySeconds` configurable

It probably makes sense to make the other probe properties configurable as well for consistency.